pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -233,6 +244,64 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonChain
 		}
 	}
 
+	// Check if a Tekton Chain Secret InstallerSet already exists, if not then create one
+	secretLabelSelector, err := common.LabelSelector(secretLs)
+	if err != nil {
+		return err
+	}
+	existingSecretInstallerSet, err := tektoninstallerset.CurrentInstallerSetName(ctx, r.operatorClientSet, secretLabelSelector)
+	if err != nil {
+		return err
+	}
+	if existingSecretInstallerSet == "" {
+		tc.Status.MarkInstallerSetNotAvailable("Chain Secret InstallerSet not available")
+		createdIs, err := r.createSecretInstallerSet(ctx, tc)
+		if err != nil {
+			return err
+		}
+		return r.updateTektonChainStatus(tc, createdIs)
```

Comment:
lgtm 
here we are overriding installer set name in status of TektonChains
there is only one field for installer set name in CR and now we have 2 for chains
may be in future we could remove that field or add all installerset names in map

Reply:
Yes, I also was worrying the same, then checked the pipeline reconciler where we have three installerset like pre, post, and main. There was also the same thing so I kept it as it is for now.

If you want i can remove this logic for now and in the future can do the changes for array/map.

Reply:
yeah if we can keep the core installer set in status for now would be okay I think
and skip the secret one

Reply:
updated the PR with this